### PR TITLE
linaro-forge: added package file for rebranded product

### PR DESCRIPTION
**Pull request by kenche-linaro**

### CODE REVIEW

**Comment by kenche-linaro** [comment]
@tldahlgren could you advise on who can review this and how we can get it merged please? Thanks.

File: var/spack/repos/builtin/packages/linaro-forge/package.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        )
+        version(
+            "23.0.2", sha256="698fda8f7cc05a06909e5dcc50b9956f94135d7b12e84ffb21999a5b45c70c74"
+        )
+        version(
+            "23.0.1", sha256="552e4a3f408ed4eb5f1bfbb83c94530ee8733579c56c3e98050c0ad2d43eb433"
+        )
+        version("23.0", sha256="7ae20bb27d539751d1776d1e09a65dcce821fc6a75f924675439f791261783fb")
+        version(
+            "22.1.4", sha256="4e2af481a37b4c99dba0de6fac75ac945316955fc4170d06e321530adea7ac9f"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/linaro-forge/package.py:37]: Do you mean `"22.0.4"` here?

File: var/spack/repos/builtin/packages/linaro-forge/package.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                version,
+                version,
+                platform.machine(),
+            )
+        else:
+            return "https://downloads.linaroforge.com/%s/linaro-forge-%s-linux-%s.tar" % (
+                version,
+                version,
+                platform.machine(),
+            )
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/linaro-forge/package.py]: FYI. You could simplify this along the lines:

```suggestion
        pre = "arm" if version < Version("23.0") else "linaro"
        return f"https://downloads.linaroforge.com/{version}/{pre}-forge-{version}-linux-{platform.machine()}.tar"
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the homepage and version sha256 for one of the platforms _except_ I think there is one incorrect version for one of the platforms.  Also made a suggestion to simplify the url conversion method.

I also wonder if you should `git mv` and update the `arm-forge` package?  Or at least flag it as "deprecated"?

**Comment by tldahlgren** [comment]
> @tldahlgren could you advise on who can review this and how we can get it merged please? Thanks.

Pinging a Spack maintainer works.  You can also post in Spack's #pull-requests slack channel though that's usually if the PR has been sitting a week or two.  (I see this one has been sitting a lot longer.)

**Comment by tldahlgren** [comment]
Adding @NickRF since he is the maintainer of the `arm-forge` package in case he can shed some light on my question about renaming/moving or "deprecating" the existing package.

**Comment by NickRF** [comment]
> I also wonder if you should `git mv` and update the `arm-forge` package? Or at least flag it as "deprecated"?

I haven't used Spack's "deprecate" functionality before, although the docs suggest it is for withdrawing packages with security issues "only in extraordinary circumstances", so that might not be a good fit.

As for moving vs copying, I think you can make a case for either. The "arm-forge" package still provides versions that "linaro-forge" does not, which provides some (admittedly limited) benefit, although the continued presence of "arm-forge" makes it more likely that users won't discover the new "linaro-forge" package/versions. (A single "forge" package could offer a solution, although I'm not sure if the older packages will continue to be available long-term)

It's also worth pointing out that "ddt" is another related outdated package, and that the "timemory" package has an optional dependency on "arm-forge", which may need updating at some point.

**Comment by NickRF** [comment]
> I also wonder if you should `git mv` and update the `arm-forge` package? Or at least flag it as "deprecated"?

I haven't used Spack's "deprecate" functionality before, although the docs suggest it is for withdrawing packages with security issues "only in extraordinary circumstances", so that might not be a good fit.

As for moving vs copying, I think you can make a case for either. The "arm-forge" package still provides versions that "linaro-forge" does not, which provides some (admittedly limited) benefit, although the continued presence of "arm-forge" makes it more likely that users won't discover the new "linaro-forge" package/versions. (A single "forge" package could offer a solution, although I'm not sure if the older packages will continue to be available long-term)

It's also worth pointing out that "ddt" is another related outdated package, and that the "timemory" package has an optional dependency on "arm-forge", which may need updating at some point.

**Review by NickRF — APPROVED** [review]

File: var/spack/repos/builtin/packages/linaro-forge/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/linaro-forge/package.py:37]: This is the blocker fix for me since I get an error trying to fetch 22.1.4.

Comment (kenche-linaro) [reply var/spack/repos/builtin/packages/linaro-forge/package.py:37]: This should be resolved now, thanks.

File: var/spack/repos/builtin/packages/linaro-forge/package.py (hunk 2 above)
Comment (kenche-linaro) [reply var/spack/repos/builtin/packages/linaro-forge/package.py]: Thanks, this is much cleaner.

File: var/spack/repos/builtin/packages/linaro-forge/package.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        )
+        version(
+            "23.0.1", sha256="08cffef2195ea96872d56e827f320eed40aaa82fd3b62d4c661a598fb2fb3a47"
+        )
+        version("23.0", sha256="0962c7e0da0f450cf6daffe1156e1f59e02c9f643df458ec8458527afcde5b4d")
+        version(
+            "22.1.3", sha256="6479c3a4ae6ce6648c37594eb0266161f06d9f89010fca9299855848661dda49"
+        )
+        version(
+            "22.0.4", sha256="f4cb5bcbaa67f9209299fe4653186a2829760b8b16a2883913aa43766375b04c"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/linaro-forge/package.py:57]: If you're supporting `22.1.4` for `aarch64` and `arm64`, do you intend to support `22.1.4` (or only `22.0.4`) for `ppc64le` and `x86_64`?

In other words, are you deliberately supporting only `22.1.4` for those machines and only `22.0.4` for the other or do you intend for all machines support the same version(s)?

Comment (kenche-linaro) [reply var/spack/repos/builtin/packages/linaro-forge/package.py:57]: We deliberately released a minor update (`22.1.4`) for the `aarch64/arm64` version. The latest `22.1.x` version for `x86_64` and `ppc64le` is `22.1.3`.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/linaro-forge/package.py:57]: Great!  Thanks for the clarification.

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**